core/taskengine/engine.go

```diff
@@ -588,8 +598,18 @@ func (n *Engine) TriggerTask(user *model.User, payload *avsproto.UserTriggerTask
 		return nil, grpcstatus.Errorf(codes.NotFound, TaskNotFoundError)
 	}
 
+	var reason *avsproto.TriggerReason
+	if payload.Reason != nil {
+		reason = payload.Reason
+	} else {
+		reason = &avsproto.TriggerReason{
+			Type: avsproto.TriggerReason_Manual,
+		}
+		n.logger.Debug("Creating default manual trigger reason", "task_id", payload.TaskId)
+	}
```

Comment:
this code looks identical to the above. We should extract this common code to a function `getTriggerReasonOrDefault`.

Also, do not use `TriggerReason_Manual` as the default value. Could we define a default TriggerReason enum?